webapp/src/modules/bid/selectors.js

```diff
@@ -0,0 +1,70 @@
+import { createSelector } from 'reselect'
+import { getAddress } from '@dapps/modules/wallet/selectors'
+import { isLoadingType } from '@dapps/modules/loading/selectors'
+
+import { BID_REQUEST, CANCEL_BID_REQUEST } from './actions'
+import { LISTING_STATUS } from 'shared/listing'
+
+export const getState = state => state.bid
+export const getData = state => getState(state).data
+export const isLoading = state => getState(state).loading.length > 0
+export const getLoading = state => getState(state).loading
+export const getError = state => getState(state).error
+
+export const isBidIdle = state => isLoadingType(getLoading(state), BID_REQUEST)
+export const isCancelIdle = state =>
+  isLoadingType(getLoading(state), CANCEL_BID_REQUEST)
+
+export const getBidById = (state, bidId, assetType, assetId) => {
+  const bids = getData(state)
+  const bid = bids[bidId]
+  return bid && bid.asset_id === assetId && bid.asset_type === assetType
+    ? bid
+    : null
+}
+
+export const getBidByAssetIdFactory = (assetType, assetId) =>
+  createSelector(getAddress, getData, (userAddress, bids) => {
+    const bidId = Object.keys(bids).filter(bidId => {
+      const bid = bids[bidId]
+      return (
+        bid &&
+        bid.asset_id === assetId &&
+        bid.asset_type === assetType &&
+        bid.bidder === userAddress
+      )
+    })[0]
+    return bidId ? bids[bidId] : null
```

Comment:
I don't understand this selector

- Why the `filter` if we're getting the first result?
- Why get only the `bidId` and then re-access the `bids[bidId]`
- Are you trying to do a `find`?